### ISSUE TITLE
Exclusion patterns work on embedded interfaces, not the actual type of the receiver

### ISSUE THREAD

**Issue by antifuchs** [kisielk/errcheck#132]
I'm trying to remove one of my least favorite errcheck highlights, `defer response.Body.Close()`, which conveniently is a Close on a `ReadCloser` - a thing that doesn't cause errors I ever care about. So I added it to my `-exclude` file, like so:

```
(io.ReadCloser).Close
```

Unfortunately, this code still makes errcheck signal there's an interesting error to be caught:

``` go
package playground

import "io"

type foo struct{}

func (f *foo) Read(p []byte) (n int, err error) {
	return 0, nil
}

func (f *foo) Close() error {
	return nil
}

func tryIt() {
	var f io.ReadCloser = &foo{}
	defer f.Close()  // Shouldn't highlight this, but it does.
}
```

Does `-exclude` not work for interfaces' methods?

**Comment by dominikh** [comment]
The problem here is that the method is really `(io.Closer).Close` due to the way embedded interfaces work. Excluding that method does work.

**Comment by antifuchs** [comment]
Hrm, that's a bit more wide-ranging than I'd like, but probably worth using considering that I would like to cut down on the number of lints that I ignore in idiomatic code. Would it be possible to issue a warning if an excluded function name can never match?

**Comment by dominikh** [comment]
I'll leave this issue open in case someone wants to work on improving the exclusion matching to look at the receiver type in a way a human would expect (i.e. io.ReadCloser in this case)

I don't think a warning is a good idea in general. People are likely to share exclusion lists for different packages or projects, and the method sets can vary.

**Comment by dtcaciuc** [comment]
I just tried out this example and it worked as expected. It looks like this was fixed in #137... Good to close?

**Comment by kisielk** [comment]
Looks like it is fixed to me, there are test cases too.
